squashfs-tools/action.c

```diff
@@ -575,11 +575,14 @@ int parse_action(char *s, int verbose)
 		spec_count = other_count ++;
 		spec_list = &other_spec;
 	}
-	
+
+	old_realloc_ptr = *spec_list;
 	*spec_list = realloc(*spec_list, (spec_count + 1) *
```

Comment:
Maybe it's better to use a macro for these realloc-or-abort sections to avoid unecessary variables.

```c
#define REALLOC_OR_ABORT(type, dest, src, size) \
  do { \
    type __old_src = (src); \
    dest = realloc((src), size); \
    if (dest == NULL) { \
      free(__old_src); \
      MEM_ERROR(); \
    } \
  } while (0)
```
or more concisely since we're aborting the program on failure anyway:

```c
#define REALLOC_OR_ABORT(dest, src, size) \
  do { \
    /* cppcheck-suppress memleakOnRealloc ; abort is called on failure */ \
    dest = realloc((src), size); \
    if (dest == NULL) { \
      MEM_ERROR(); \
    } \
  } while (0)
```